controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_types.go

```diff
@@ -99,6 +93,12 @@ type KubeadmControlPlaneMachineTemplate struct {
 	// InfrastructureRef is a required reference to a custom resource
 	// offered by an infrastructure provider.
 	InfrastructureRef corev1.ObjectReference `json:"infrastructureRef"`
+
+	// NodeDrainTimeout is the total amount of time that the controller will spend on draining a controlplane node
+	// The default value is 0, meaning that the node can be drained without any time limitations.
+	// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
+	// +optional
+	NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
```

Comment:
there are e2e templates that need to be updated https://github.com/kubernetes-sigs/cluster-api/blob/master/test/e2e/data/infrastructure-docker/v1alpha4/cluster-template-node-drain/cluster-with-kcp.yaml#L8

Also, this should be added to the providers alpha4 docs since it requires updating templates

Reply:
Done !